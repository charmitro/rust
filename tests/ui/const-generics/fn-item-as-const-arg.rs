// src/test/ui/const-generics/fn-item-as-const-arg.rs

#![feature(min_generic_const_args)] //~ WARNING the feature `min_generic_const_args` is incomplete

fn func() {}

fn test<F>() where [(); func]: {}
//~^ ERROR function items cannot be used as const arguments

fn main() {}
