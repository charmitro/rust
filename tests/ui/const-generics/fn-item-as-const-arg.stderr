warning: the feature `min_generic_const_args` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/fn-item-as-const-arg.rs:3:12
   |
LL | #![feature(min_generic_const_args)]
   |            ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #132980 <https://github.com/rust-lang/rust/issues/132980> for more information
   = note: `#[warn(incomplete_features)]` on by default

error: function items cannot be used as const arguments
  --> $DIR/fn-item-as-const-arg.rs:7:25
   |
LL | fn test<F>() where [(); func]: {}
   |                         ^^^^

error: aborting due to 1 previous error; 1 warning emitted

